clippy_lints/src/casts/unnecessary_cast.rs

```diff
@@ -30,8 +30,10 @@ pub(super) fn check<'tcx>(
         }
     }
 
+    let cast_str = snippet_opt(cx, cast_expr.span).unwrap_or_default();
```

Comment:
nit, use [clippy_utils](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/index.html)::[source](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/index.html)::[snippet](https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/source/fn.snippet.html#) instead (with canonical fallback being `..`, as shown in examples

Reply:
The original code uses `snippet_opt,` and the only thing I do is move the variable; there are also some other `snippet_opt` usages in the file, so I'm not sure if I should update them all in the PR. It seems a little off-topic.